File: bubblesort.c
#include<stdio.h>
void bubblesort(int arr[],int n){
    for(int i=0;i<n-1;i++){
        for(int j=0;j<n-i-1;j++){
            if(arr[j]>arr[j+1]){
                int temp=arr[j+1];
                arr[j+1]=arr[j];
                arr[j]=temp;
            }
        }
    }
}
void print(int arr[],int n){
    for(int i=0;i<n;i++){
        printf("%d ",arr[i]);       
    }
    printf("\n");
}
int main(){
    int arr[]={15,16,45,2,3,55};
    int n=sizeof(arr)/sizeof(arr[0]);
    printf("original array:");
    print(arr,n);
    bubblesort(arr,n);
    printf("sorted araay:");
    print(arr,n);
    return 0;
}

File: bubblesort.java
import java.util.*;
public class bubblesort {
    public static void sort(int nums[]){
        for(int i=0;i<nums.length-1;i++){
            for(int j=0;j<nums.length-1-i;j++){
                if(nums[j]>nums[j+1]){
                    int temp=nums[j+1];
                    nums[j+1]=nums[j];
                    nums[j]=temp;
                }
            }
        }
    }
    public static void print(int nums[]){
        for(int i=0;i<nums.length;i++){
            System.out.print(nums[i]+" ");
        }
        System.out.println();
    }
    public static void main(String[] args) {
        int nums[]={5,4,1,3,2};
        sort(nums);
        print(nums);
    }
}
